GMG/GMG/UI/Components/Metronome.swift

```diff
@@ -6,13 +6,149 @@
+        }
+        
+        if let url = Bundle.main.url(forResource: "Metronome_Low", withExtension: "wav") {
+            lowPlayer = try? AVAudioPlayer(contentsOf: url)
+            lowPlayer?.volume = 1.0
+            lowPlayer?.prepareToPlay()
+        } else {
+            print("[Metronome] Missing resource: Metronome_Low.wav (check target membership)")
+        }
+    }
+
+    // 소리 중 강박을 구분을 위한 인텍스 설정 (3/4, 4/4 박자는 첫 번째 박만 강박이지만, 6/8 박자의 경우 1, 4번 박에 강박을 줘야 함) *강박: Metronome 재생 중 기준을 잡는 높은 음
+    private func shouldUseHigh(beatIndex: Int) -> Bool {
+        if selectedTimeSig.numerator == 6 && selectedTimeSig.denumerator == .eighth {
+            return beatIndex == 0 || beatIndex == 3
+        } else {
+            return beatIndex == 0
+        }
+    }
+
+    // 위에서 구분한 Index에 따라 박자별 High or Low 음성 재생
+    private func playClick(for beatIndex: Int) {
+        if shouldUseHigh(beatIndex: beatIndex) {
+            guard let audioPlayer = highPlayer else {
+                print("[Metronome] highPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        } else {
+            guard let audioPlayer = lowPlayer else {
+                print("[Metronome] lowPlayer is nil")
+                return
+            }
+            audioPlayer.currentTime = 0
+            audioPlayer.play()
+        }
+    }
+
     var body: some View {
-        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
+        
+        //TimelineView를 사용해 beatInterval마다 뷰가 업데이트 되도록 구성함
+        TimelineView(.periodic(from: .now, by: beatInterval)) { context in
+            BeatBar(beatsPerBar: beatsPerBar, currentBeat: currentBeatToShow)
+            .animation(.spring(response: 0.28, dampingFraction: 0.85, blendDuration: 0.2), value: beatsPerBar)
+            .padding()
+            .onChange(of: context.date) { _, _ in
+                guard isPlaying, let start = startDate else { return }
+                let step = Int(floor(Date().timeIntervalSince(start) / beatInterval))
+                if step != lastStepPlayed {
+                    lastStepPlayed = step
+                    playClick(for: currentBeatToShow)
+                }
+            }
+        }
+        .onAppear {
+            configureAudioSession()
+            loadPlayers()
+        }
+        //박자가 변경될 때 자연스럽게 CD 갯수가 바뀌는 애니메이션
+        .onChange(of: selectedTimeSig) { _, _ in
+            withAnimation(.spring(response: 0.28, dampingFraction: 0.85, blendDuration: 0.2)) {
+                if isPlaying { startDate = Date(); lastStepPlayed = -1 }
+            }
+        }
+        .onChange(of: isPlaying) { _, newValue in
+            if newValue {
+                startDate = Date()
+                lastStepPlayed = -1
+            }
+        }
+        .onChange(of: bpm.value) { _, _ in
+            if isPlaying { startDate = Date(); lastStepPlayed = -1 }
+        }
```

Comment:
오디오 관련 메모리 해제 로직이 필요해보입니더
```suggestion
    .onDisappear() {
        // audio session 관련 메모리 해제 로직
        // 1. 플레이어 중지 및 메모리에서 해제
        highPlayer?.stop()
        lowPlayer?.stop()
        highPlayer = nil
        lowPlayer = nil
        
        // 2. 오디오 세션 비활성화
        do {
            try AVAudioSession.sharedInstance().setActive(false, options: .notifyOthersOnDeactivation)
        } catch {
            print("[Metronome] AudioSession deactivation error: \(error)")
        }
    }
```